ape_trezor/_cli.py

```diff
@@ -1,120 +1,145 @@
+    callback=lambda ctx, param, arg: HDBasePath(arg),
 )
-@click.argument("alias")
-def delete(alias):
-    if alias not in container.aliases:
-        raise Abort(f"Account with alias '{alias}' does not exist")
+def add(cli_ctx, alias, hd_path):
+    """Add a account from your Trezor hardware wallet"""
+
+    client = TrezorClient(hd_path)
+    choices = AddressPromptChoice(client, hd_path)
+    address, account_hd_path = choices.get_user_selected_account()
+    container = accounts.containers.get("trezor")
+    container.save_account(alias, address, str(account_hd_path))
+    cli_ctx.logger.success(f"Account '{address}' successfully added with alias '{alias}'.")
+
+
+@cli.command()
+@ape_cli_context()
+@existing_alias_argument(account_type=TrezorAccount)
+def delete(cli_ctx, alias):
+    """Remove a Trezor account from your ape configuration"""
+
+    container = accounts.containers.get("trezor")
+    container.delete_account(alias)
+    cli_ctx.logger.success(f"Account '{alias}' has been removed.")
+
+
+@cli.command()
+@ape_cli_context()
+@skip_confirmation_option("Don't ask for confirmation when removing all accounts")
+def delete_all(cli_ctx, skip_confirmation):
+    """Remove all trezor accounts from your ape configuration"""
+
+    container = accounts.containers.get("trezor")
+    trezor_accounts = _get_trezor_accounts()
+    if len(trezor_accounts) == 0:
+        cli_ctx.logger.warning("No accounts found.")
+        return
 
-    path = container.data_folder.joinpath(f"{alias}.json")
-    try:
-        path.unlink()
-        notify("SUCCESS", f"Account '{alias}' has been removed")
-    except Exception as e:
-        raise Abort(f"File does not exist: {path}") from e
+    user_agrees = skip_confirmation or click.confirm("Remove all Trezor accounts from ape?")
+    if not user_agrees:
+        cli_ctx.logger.info("No account were removed.")
+        return
+
+    for account in trezor_accounts:
+        container.delete_account(account.alias)
+        cli_ctx.logger.success(f"Account '{account.alias}' has been removed.")
 
 
 @cli.command(short_help="Sign a message with your Trezor device")
 @click.argument("alias")
 @click.argument("message")
-def sign_message(alias, message):
+@ape_cli_context()
+def sign_message(cli_ctx, alias, message):
+
     if alias not in accounts.aliases:
-        notify("ERROR", f"Account with alias '{alias}' does not exist")
+        cli_ctx.logger.warning(f"Account with alias '{alias}' does not exist.")
         return
 
-    eip191message = SignableMessage(
-        version=b"E",
-        header=f"thereum Signed Message:\n{len(message)}".encode("utf8"),
-        body=message.encode("utf8"),
-    )
+    eip191message = encode_defunct(text=message)
     account = accounts.load(alias)
     signature = account.sign_message(eip191message)
-    notify("Signature:", signature.encode_vrs().hex())
+    signature_bytes = signature.encode_vrs()
```

Comment:
(We are discussing offline but making a note here)

`Account.recover_message(eip191message, signature=signature_bytes)` asks for the signature in `rsv` mode. Yet, we are passing it in `vrs` mode... This `sign-message` seems to work as-is. However, I tried signing a typed-message and it accepted the correct way, the `rsv` and not the `vrs`

Reply:
Updated with a234e18